### PR TITLE
refactor(ansible/docker): split docker roles

### PR DESCRIPTION
**Pull request by kenji-miyake**
## Description

I split the `docker` role into multiple for the following reasons:
- To make the role of each Ansible role clear
- To enable people to select which features to install.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [x] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [x] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: ansible/playbooks/core.yaml [hunk 1]
```diff
@@ -8,7 +8,6 @@
           - rmw_implementation: "{{ rmw_implementation }}"
   roles:
     - role: autoware.dev_env.autoware_core
-    - role: autoware.dev_env.docker
```
Comment (kenji-miyake) [inline ansible/playbooks/core.yaml:11]: I removed the `docker` role from `core.yaml` and `universe.yaml` because they don't have to depend on the role.

**Review by esteve — DISMISSED** [review]
@kenji-miyake these changes make the code more readable, thanks!

**Comment by kenji-miyake** [comment]
I should have fixed a link.
image

**Comment by kenji-miyake** [comment]
@esteve Could you check my new commits? :pray:

**Review by esteve — APPROVED** [review]

**Pull request merged**